src/lib/yang/schema.lua

```diff
@@ -700,10 +707,11 @@ function resolve(schema, features)
                   node.body[k] = v
                end
                -- TODO: Handle refine and augment statements.
-            else
-               node.body[k] = visit(v, env)
             end
          end
+         for k,v in pairs(node.body or {}) do
```

Comment:
Ah you are right, you just moved the visit down into a separate loop; ok.  I thought you duplicated the loop.  So, not quadratic; however not quite right either as I think you don't want to revisit a node which has been included here via a grouping/uses.  I think that would be incorrect as it would try to resolve types and things a second time, when they were already resolved before, and it would do so from the wrong environment.  I am not sure that visiting is idempotent, and I think it is a good idea to avoid doing the typedef/grouping part of `visit` twice for any given node.  For this reason (and not the quadratic reason) I still think a post-pass that just propagates the `config` flag is the right thing.